### DISCUSSION TITLE
Changes to layout etc

### DISCUSSION THREAD

**Discussion by Steph1116** [jake1164/Pico-RGB-Matrix-Weather-Clock#25] [General]
I'm loving this addition from the original clock and date setup. 

The main reason I got the 64x32 clock was to replace my Alexa show 5. I wanted to customize the time and be able to display the date, weather, and so on. My biggest draw to pico w, coding and the rgb panel was to be able to pick what colours that  would be displayed. Red being my main choice so I could have it show the time and so on at night in my blacked out room without the screen lighting up the room.

So with that, this design is great, but a starter to where I'd like to be for the final product sorta speak. 

I'd like the weather icons to do their same function but all red icons. I'd like to adjust the size of the temperature and icons to an 18x18. And then 18 pixels height for each line/row. This would then add spacing of 3 from the top, 2 middle 2, middle, 3 pixel to the bottom between the lines/rows.

The temperature and icon on the first line, clock second line and the scrolling text on the third.

I'd then try to get the whole thing to be colourful during the day, say 9am-9pm and then all red 9pm-9am. 

But I'm a beginner, so most, if not all of that will take time.

**Comment by jake1164** [comment]
Hi,

The red at night is an interesting idea.. sorta a dark mode if you will.


________________________________
From: Steph1116 ***@***.***>
Sent: Monday, March 13, 2023 12:01 PM
To: jake1164/Pico-RGB-Matrix-Clock ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [jake1164/Pico-RGB-Matrix-Clock] Changes to layout etc (Discussion #25)


I'm loving this addition from the original clock and date setup.

The main reason I got the 64x32 clock was to replace my Alexa show 5. I wanted to customize the time and be able to display the date, weather, and so on. My biggest draw to pico w, coding and the rgb panel was to be able to pick what colours that would be displayed. Red being my main choice so I could have it show the time and so on at night in my blacked out room without the screen lighting up the room.

So with that, this design is great, but a starter to where I'd like to be for the final product sorta speak.

I'd like the weather icons to do their same function but all red icons. I'd like to adjust the size of the temperature and icons to an 18x18. And then 18 pixels height for each line/row. This would then add spacing of 3 from the top, 2 middle 2, middle, 3 pixel to the bottom between the lines/rows.

The temperature and icon on the first line, clock second line and the scrolling text on the third.

I'd then try to get the whole thing to be colourful during the day, say 9am-9pm and then all red 9pm-9am.

But I'm a beginner, so most, if not all of that will take time.

**Comment by Steph1116** [comment]
Correct. With the Amazon show 5 I had it set to white during the day and red at 9pm for overnight. The big issue I had was the screen was an lcd and it had a semi dark background. So it still lit up my room in a faint glow. More less then if the time was white though.

**Comment by Steph1116** [comment]
The bmp icons you use, do they have solid colour icons? Like i mentioned, I’m looking to just have them display red.

**Reply by jake1164** [reply]
I think this could be achieved using the displayIO settings.
https://docs.circuitpython.org/en/latest/shared-bindings/displayio/index.html
Specifically the colorspace in the colorconverter.

**Comment by jake1164** [comment]
@Steph1116 I renamed the repo slightly.

**Comment by Steph1116** [comment]
I've been searching for red weather icons to swap out the current batch for overnight display. No luck as of yet. Let me know if you stumble upon any. 

With the current code, could we make the temperature and bmp icon a bit smaller and then move the time a line higher so the scrolling text doesn't overlap as it goes by?

**Comment by jake1164** [comment]
The temperature is harder than it sounds because its font dependent. So the current is a compromise for now.  I am hopeful to move things around a bit more but have been focusing on other things like tracking down the random crash and adding persistent settings  that survive a reboot.

With the settings update I have the underpinnings to add a night mode and try to change the color to red when its dark. I think I can change the color of the icons to be dark programmatically without a second set of icons.
________________________________

**Comment by Steph1116** [comment]
No problem. I highly appreciate all the hard work your doing with this code. I wish I had a second 64x32 clock. Unfortunately they are $80 plus right now from waveshare/Amazon. So my current daily use is the original code you had with the date/time/day. Turned it all red. I have a second pico w with this code and I just swap the pico w when I want to play around with it. I have a 64x64 panel with the interstate 75w that I haven't done anything with yet. Unfortunately this code doesn't work with it without a ton of changes. But anyhow, looking forward to apply your changes tonight and see the results. I also wish I could contribute, but I'm no where near your skill level on this stuff.

**Comment by jake1164** [comment]
Well having someone else to run the code is a huge help in itself!

I picked up this 64x64 with hopes to setting it up with the same code base.  In theory the waveshare board should just plug into the bigger panel.  I have not tried it yet since I have enough to keep me busy but its on the horizon.

https://www.amazon.com/Individual-Adjustable-Brightness-Displaying-Compatible/dp/B0B65J1D9Q/ref=sr_1_2?crid=2WAV5PDSU15GR&keywords=waveshare+64x64&qid=1679490399&sprefix=waveshare+64x%2Caps%2C625&sr=8-2
________________________________

**Comment by Steph1116** [comment]
Perhaps look into grabbing the interstate 75 w. It's not too much. I have played around with it a bit on the 64x64 and it seems far simpler to throw something together with it then the 64x32 matrix clock setup. I personally do not use the remote or buttons on the side of the unit, nor do I use the alarm/buzzer. It's all bulk extra to me. I'm focused completely on the display and that's why I originally bought the pico w and the 64x64. As I began looking for codes online, I wasn't able to find anything suitable for my project. So I then grabbed the 64x32 clock. And here I am today...

**Comment by Steph1116** [comment]
I'm going to put the code back on my clock tonight or tomorrow to test out your fixes. I'll let you know if I encounter anything. Any thoughts on the interstate 75 w board? I'm hoping to get good enough with code to make something with my interstate 75 for my 64x64 to display and then use the 75w with the 64x32 as a bedroom clock.

**Comment by jake1164** [comment]
It looks very interesting.  The only place I could find it in stock however was in UK and the shipping would put it way out of range.
________________________________

**Comment by Steph1116** [comment]
Ah. They usually go in stock and get picked up quickly for some odd reason. Took some time to scoop up mine.

**Comment by Steph1116** [comment]
Got the weather clock up and running. Had a bit of an issue with the dimming when I first launched the code. My room is pretty dark all the time. I simply just changed the code to false to disable the dimming. 

What do you think of changing the scrolling "****-**-**" date format to "Sat, Mar 23rd" format, and also how difficult or easy would it be? I'd say the short form of the day and month so it's sits nicely on screen.

**Comment by Steph1116** [comment]
Just an update that the code is holding strong. For my daily use I haven't had any issues.
